packages/lambdas/src/patient-discovery.ts

```diff
@@ -1,103 +1,42 @@
+const apiUrl = getEnvVarOrFail("API_URL");
+const mpi = new MPIMetriportAPI(apiUrl);
 
-// Function to extract necessary fields and construct the responses
-async function processRequest(
-  payload: PatientDiscoveryRequestIncoming
-): Promise<PatientDiscoveryResponseOutgoing> {
-  // validate with zod schema
-  const baseRequest = baseRequestSchema.parse({
-    id: payload.id,
-    timestamp: payload.timestamp,
-    samlAttributes: payload.samlAttributes,
-    patientId: payload.patientId,
-  });
-
-  const fullRequest: PatientDiscoveryRequestIncoming = {
-    ...baseRequest,
-    patientResource: payload.patientResource,
-  };
-
-  if (Math.random() > 0.5) {
-    return constructErrorResponse(fullRequest);
+export const handler = Sentry.AWSLambda.wrapHandler(async (event: APIGatewayProxyEvent) => {
+  if (!event.body) {
+    return buildResponse(400, "Request body is missing");
   }
-  return constructSuccessResponse(fullRequest, true);
-}
-
-// Function to construct error response
-function constructErrorResponse(
-  payload: PatientDiscoveryRequestIncoming
-): PatientDiscoveryResponseOutgoing {
-  return {
-    id: payload.id,
-    timestamp: payload.timestamp,
-    responseTimestamp: new Date().toISOString(),
-    patientMatch: false,
-    xcpdHomeCommunityId: payload.samlAttributes.homeCommunityId,
-    operationOutcome: {
-      resourceType: "OperationOutcome",
+  try {
+    const payload: PatientDiscoveryRequestIncoming = JSON.parse(event.body);
+    const baseRequest = baseRequestSchema.parse({
       id: payload.id,
-      issue: [
-        {
-          severity: "error",
-          code: "structure",
-          details: { text: "Invalid structure" },
-        },
-      ],
-    },
-  };
-}
+      timestamp: payload.timestamp,
+      samlAttributes: payload.samlAttributes,
+      patientId: payload.patientId,
+    });
 
-// Function to construct success response
-function constructSuccessResponse(
-  payload: PatientDiscoveryRequestIncoming,
-  patientMatch: boolean
-): PatientDiscoveryResponseOutgoing {
-  const patientResource: Patient = {
-    resourceType: "Patient",
-    identifier: [
-      {
-        system: "http://hl7.org/fhir/sid/us-ssn",
-        value: "666-10-0001",
-      },
-    ],
-    name: [
-      {
-        family: "NWHINZZZTESTPATIENT",
-        given: ["NWHINONE"],
-      },
-    ],
-    gender: "male",
-    birthDate: "1981-01-01",
-    address: [
-      {
-        line: ["1100 Test Street"],
-        city: "Helena",
-        state: "AL",
-        postalCode: "35080",
-        country: "USA",
-      },
-    ],
-  };
+    const fullRequest: PatientDiscoveryRequestIncoming = {
+      ...baseRequest,
+      patientResource: payload.patientResource,
+    };
+
+    const result = await processIncomingRequest(fullRequest, mpi);
+    return buildResponse(200, result);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (err: any) {
+    return buildResponse(400, err.message);
```

Comment:
This is assuming the error was on the client side, but it can be an "internal error", no?

Reply:
We don't return internal errors as errors. In the `processingIncomingRequest` we catch all errors and create a structured response object. So we should only have 400 errors, and there shouldnt even be a try catch block there, which I will remove

Reply:
So right now we have 400s being returned if they request is missing the request body or does not match the expected input schema. That seems appropriate to me since all of those are client side issues.